libretroshare/src/retroshare/rsmsgs.h

```diff
@@ -93,9 +95,31 @@ const ChatLobbyFlags RS_CHAT_LOBBY_FLAGS_PGP_SIGNED    ( 0x00000010 ) ; // requi
 typedef uint64_t	ChatLobbyId ;
 typedef uint64_t	ChatLobbyMsgId ;
 typedef std::string ChatLobbyNickName ;
-typedef std::string RsMailMessageId;			// should be uint32_t !!
-typedef uint64_t    MessageId ;
+typedef std::string RsMailMessageId; // TODO: rebase on t_RsGenericIdType
 
+/**
+ * Used to return a tracker id so the API user can keep track of sent mail
+ * status
+ */
+struct RsMailTrackId : RsSerializable
```

Comment:
Given the next comment, I suppose that it would be even simpler to define that ID as a t_RSGenericId<some const>. You'd get serialization, comparisons, and type checking for free.

Reply:
see my reply upster